### PR TITLE
libgit2: fix cmake options on FreeBSD

### LINKED ISSUE #22745: Pkg.update() fails on FreeBSD due to libgit2 issues
```
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.7.0-DEV.887 (2017-07-10 18:22 UTC)
 _/ |\__'_|_|_|\__'_|  |  Commit 86395faee1* (0 days old master)
|__/                   |  x86_64-unknown-freebsd11.0

julia> Pkg.update()
INFO: Updating METADATA...
ERROR: METADATA cannot be updated. Resolve problems manually in /home/alex/.julia/v0.7/METADATA.
GitError(Code:ERROR, Class:SSL, there is no TLS stream available)
macro expansion at ./libgit2/error.jl:99 [inlined]
#fetch#51(::Base.LibGit2.FetchOptions, ::String, ::Function, ::Base.LibGit2.GitRemote, ::Array{AbstractString,1}) at ./libgit2/remote.jl:291
(::Base.LibGit2.#kw##fetch)(::Array{Any,1}, ::Base.LibGit2.#fetch, ::Base.LibGit2.GitRemote, ::Array{AbstractString,1}) at ./<missing>:0
#fetch#95(::String, ::String, ::Array{AbstractString,1}, ::Nullable{Base.LibGit2.AbstractCredentials}, ::Function, ::Base.LibGit2.GitRepo) at ./libgit2/libgit2.jl:277
(::Base.Pkg.Entry.##33#39)(::Base.LibGit2.GitRepo) at ./pkg/entry.jl:371
with at ./libgit2/types.jl:775 [inlined]
with at ./libgit2/types.jl:781 [inlined]
update(::String, ::Set{String}) at ./pkg/entry.jl:354
#4 at ./pkg/dir.jl:36 [inlined]
cd(::Base.Pkg.Dir.##4#7{Array{Any,1},Base.Pkg.Entry.#update,Tuple{String,Set{String}}}, ::String) at ./file.jl:70
#cd#1(::Array{Any,1}, ::Function, ::Function, ::String, ::Vararg{Any,N} where N) at ./pkg/dir.jl:36
update() at ./pkg/pkg.jl:228
eval(::Module, ::Expr) at ./repl/REPL.jl:3
eval_user_input(::Any, ::Base.REPL.REPLBackend) at ./repl/REPL.jl:67
macro expansion at ./repl/REPL.jl:98 [inlined]
(::Base.REPL.##1#2{Base.REPL.REPLBackend})() at ./event.jl:73
Stacktrace:
 [1] (::Base.Pkg.Entry.##33#39)(::Base.LibGit2.GitRepo) at ./pkg/entry.jl:378
 [2] with at ./libgit2/types.jl:775 [inlined]
 [3] with at ./libgit2/types.jl:781 [inlined]
 [4] update(::String, ::Set{String}) at ./pkg/entry.jl:354
 [5] #4 at ./pkg/dir.jl:36 [inlined]
 [6] cd(::Base.Pkg.Dir.##4#7{Array{Any,1},Base.Pkg.Entry.#update,Tuple{String,Set{String}}}, ::String) at ./file.jl:70
 [7] #cd#1(::Array{Any,1}, ::Function, ::Function, ::String, ::Vararg{Any,N} where N) at ./pkg/dir.jl:36
 [8] update() at ./pkg/pkg.jl:228
```
May be related to the recent libgit2 upgrade. If so, we may need to check whether there are any libgit2 patches in the FreeBSD Ports system that we'll need to pull in. EDIT: The libgit2 port hasn't been updated for the newest version yet.

### PR DESCRIPTION
**Pull request by iblislin**
Fix #22745

### CODE REVIEW

**Review by ararslan — APPROVED** [review]
I can confirm that this fixes the issue on FreeBSD. Thanks!

File: deps/libgit2.mk [hunk 1]
```diff
@@ -61,6 +58,11 @@ CERTFILE=$(OPENSSL_DIR)/certs/ca.pem
 endif
 endif # Linux
 
+ifeq ($(OS),FreeBSD)
+OPENSSL_DIR=$(shell openssl version -d | cut -d '"' -f 2)
+CERTFILE=$(OPENSSL_DIR)/cert.pem
+endif # FreeBSD
+
```
Comment (ViralBShah) [inline deps/libgit2.mk]: Shouldn't this work on linux too?

Comment (iblis17) [inline deps/libgit2.mk]: There is a block for Linux already: https://github.com/JuliaLang/julia/pull/22754/files#diff-a01ad0cac7aa5a648c5d00ec39bd9afcR44

Comment (ViralBShah) [inline deps/libgit2.mk]: I meant I was hoping to have a common block. Feel free to ignore me...

Comment (iblis17) [inline deps/libgit2.mk]: 👌

Comment (iblis17) [inline deps/libgit2.mk]: please checkout https://github.com/JuliaLang/julia/pull/22754/commits/3f93719553dc26b803dcaf9469a297d392dd522b#diff-a01ad0cac7aa5a648c5d00ec39bd9afcR44

**Pull request merged**